src/GitVersionTask/VersionAndBranchFinder.cs

```diff
@@ -27,17 +27,20 @@ public static VersionVariables GetVersion(string directory, Authentication authe
         var gitDir = GitDirFinder.TreeWalkForDotGitDir(directory);
         using (var repo = RepositoryLoader.GetRepo(gitDir))
         {
-            var ticks = DirectoryDateFinder.GetLastDirectoryWrite(directory);
-            var key = string.Format("{0}:{1}:{2}", repo.Head.CanonicalName, repo.Head.Tip.Sha, ticks);
+            var ticks = DirectoryDateFinder.GetLastDirectoryWrite(gitDir);
+            var key = string.Format("{0}:{1}:{2}",gitDir, repo.Head.CanonicalName, repo.Head.Tip.Sha);
```

Comment:
The dictionary should probably be changed to a [`ConcurrentDictionary`](https://msdn.microsoft.com/en-us/library/dd287191.aspx) too, to avoid race conditions and concurrency issues if (and with Roslyn; when) the build is parallelized.